mindmeld/resource_loader.py

```diff
@@ -400,12 +564,24 @@ def flatten_query_tree(query_tree):
         Returns:
             list: A list of Query objects.
         """
-        flattened = []
+        flattened = ProcessedQueryList()
         for _, intent_queries in query_tree.items():
             for _, queries in intent_queries.items():
-                flattened.extend(queries)
+                if not flattened.cache:
+                    flattened.cache = queries.cache
+                elif flattened.cache != queries.cache:
+                    logger.error('query_tree is built from incompatible query_caches')
+                    raise ValueError()
```

Comment:
We can add the `'query_tree is built from incompatible query_caches'` to the `ValueError `